test/heapwatch/heapWatch.py

```diff
@@ -64,15 +70,24 @@ def do_graceful_stop(signum, frame):
 signal.signal(signal.SIGTERM, do_graceful_stop)
 signal.signal(signal.SIGINT, do_graceful_stop)
 
+def jsonable(ob):
+    if isinstance(ob, bytes):
+        return base64.b64encode(ob).decode()
+    if isinstance(ob, list):
+        return [jsonable(x) for x in ob]
+    if isinstance(ob, dict):
+        return {jsonable(k):jsonable(v) for k,v in ob.items()}
+    return ob
 
 class algodDir:
-    def __init__(self, path):
+    def __init__(self, path, net=None, token=None, admin_token=None):
         self.path = path
         self.nick = os.path.basename(self.path)
-        net, token, admin_token = read_algod_dir(self.path)
         self.net = net
         self.token = token
         self.admin_token = admin_token
+        if net is None:
+            net, token, admin_token = read_algod_dir(self.path)
```

Comment:
Why is this after setting the self (net, token, admin_token)?

Reply:
good catch. fixed.